packages/job-worker/src/playout/upgrade.ts

```diff
@@ -67,13 +69,36 @@ export async function handleBlueprintUpgradeForStudio(context: JobContext, _data
 			}),
 		])
 	)
+	const routeSets = Object.fromEntries(
+		Object.entries<unknown>(result.routeSets ?? {}).map((dev) => [
+			dev[0],
+			literal<Complete<StudioRouteSet>>({
+				name: (dev[1] as StudioRouteSet).name ?? '',
+				active: (dev[1] as StudioRouteSet).active ?? false,
+				defaultActive: (dev[1] as StudioRouteSet).defaultActive ?? false,
+				behavior: (dev[1] as StudioRouteSet).behavior ?? {},
+				exclusivityGroup: (dev[1] as StudioRouteSet).exclusivityGroup ?? undefined,
+				routes: (dev[1] as StudioRouteSet).routes,
+			}),
+		])
+	)
+	const routeSetExclusivityGroups = Object.fromEntries(
+		Object.entries<unknown>(result.routeSetExclusivityGroups ?? {}).map((dev) => [
+			dev[0],
+			literal<Complete<StudioRouteSetExclusivityGroup>>({
+				name: (dev[1] as StudioRouteSetExclusivityGroup).name,
+			}),
+		])
+	)
 
 	await context.directCollections.Studios.update(context.studioId, {
 		$set: {
 			'mappingsWithOverrides.defaults': translateMappings(result.mappings),
 			'peripheralDeviceSettings.playoutDevices.defaults': playoutDevices,
 			'peripheralDeviceSettings.ingestDevices.defaults': ingestDevices,
 			'peripheralDeviceSettings.inputDevices.defaults': inputDevices,
+			'routeSets.defaults': routeSets,
```

Comment:
these paths are wrong

Reply:
Basically these PR was made prior to deciding to move to PR's directly from branch to upstream. So this PR should have been closed.
The continuation of the changes was then done in ab-dynamic-pool-assignment as a full package for merging upstream.

In retrospect this may not be optimal, but it was expected to be merged just after this and ab-dynamic-pool-assignment had been merged.

Reply:
Fixed